File: java/src/ChanghunHan/baekjoon/BOJ_16204.java
package ChanghunHan.baekjoon;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

import static java.lang.Integer.parseInt;

/**
 * @see <a href="https://www.acmicpc.net/problem/16204">
 * https://www.acmicpc.net/problem/16204
 * </a>
 */
public class BOJ_16204 {
    
    public static void main (String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = parseInt(st.nextToken());
        int M = parseInt(st.nextToken());
        int K = parseInt(st.nextToken());
        System.out.println(Math.min(M, K) + Math.min(N - M, N - K));
        br.close();
    }
    
}

File: Hyunseok-Kim/baekjoon/python/boj_16204.py
# https://www.acmicpc.net/problem/16204

n, m, k = map(int, input().split())

# 카드가 n개 중에 m개는 o 나머지는 x
# k개 o, n-k개는 x 했을 때 최대

result = k if m > k else m
result += n-k if n-m > n-k else n-m

print(result)